app/src/main/java/org/p2p/wallet/history/model/rpc/RpcHistoryTransaction.kt

```diff
@@ -323,6 +323,102 @@ sealed class RpcHistoryTransaction(
         fun getFormattedAmount(): String? = amount.totalInUsd?.asUsdTransaction(getSymbol(isStake))
     }
 
+    @Parcelize
+    data class WormholeReceive(
+        override val signature: String,
+        override val date: ZonedDateTime,
+        override val blockNumber: Int,
+        override val status: HistoryTransactionStatus,
+        override val type: RpcHistoryTransactionType,
+        val amount: RpcHistoryAmount,
+        val tokenSymbol: String,
+        val iconUrl: String?,
+        val fees: List<RpcFee>?,
+    ) : RpcHistoryTransaction(date, signature, blockNumber, status, type) {
+        @IgnoredOnParcel
+        val isBurn: Boolean
+            get() = type == RpcHistoryTransactionType.WORMHOLE_RECEIVE
+
+        @StringRes
+        fun getTitle(): Int = if (isBurn) R.string.common_burn else R.string.common_mint
+
+        fun getUsdAmount(): String = "${getFormattedAmount()}"
```

Comment:
value will be a string `"null"` if getFormattedAmount() returns null
and also it would be better to name this function `getFormattedUsdAmount` instead, imho it's more clear